stac_fastapi/types/stac_fastapi/types/clients/async_core.py

```diff
@@ -0,0 +1,592 @@
+
+        extension_schemas = [
+            schema.schema_href for schema in self.extensions if schema.schema_href
+        ]
+        catalog_children = [
+            browseable_catalog_page(
+                child,
+                base_url,
+                child["catalog_id"],
+                self.stac_version,
+                self.conformance_classes(),
+                extension_schemas,
+            )
+            for child in selected_catalog["children"]
+            if "catalog_id" in child
+        ]
+        child_collection_ids = [
+            child["collection_id"]
+            for child in selected_catalog["children"]
+            if "collection_id" in child
+        ]
+        all_collections = await self.all_collections(**kwargs)
+        collection_children = [
+            coll
+            for coll in all_collections["collections"]
+            if coll["id"] in child_collection_ids
+        ]
+
+        links = [
+            {
+                "rel": Relations.root.value,
+                "type": MimeTypes.json,
+                "href": base_url,
+            },
+            {
+                "rel": Relations.parent.value,
+                "type": MimeTypes.json,
+                "href": base_url,
+            },
+            {
+                "rel": Relations.self.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, "children"),
+            },
+        ]
+        return stac_types.Children(
+            children=catalog_children + collection_children, links=links
+        )
+
+    async def post_catalog_search(
+        self, search_request: Search, **kwargs
+    ) -> stac_types.ItemCollection:
+        """Catalog refined search (POST).
+
+        Called with `POST /search`.
+
+        Args:
+            search_request: search request parameters.
+
+        Returns:
+            ItemCollection containing items which match the search criteria.
+        """
+        # Pydantic is fine for specifying body parameters but proves difficult to
+        # use for Path parameters. Here, the starlette request is inspected instead
+        request_path = kwargs["request"]["path"]
+        split_path = request_path.split("/")[2:-1]
+        remaining_hierarchy = self.hierarchy_definition.copy()
+        # PGSTAC TODO: Add optional 'catalog' parameter to search, which will determine which
+        # collections will be included in search
+        selected_catalog = find_catalog(remaining_hierarchy, split_path)
+        if not selected_catalog:
+            raise HTTPException(
+                status_code=404, detail=f"Catalog at {'/'.join(split_path)} not found"
+            )
+        child_collections = [
+            node["collection_id"]
+            for node in selected_catalog["children"]
+            if "collection_id" in node
+        ]
+        search_request.collections = child_collections
+        return await self.post_search(search_request, **kwargs)
```

Comment:
I'm glad this is just a wrapper on top of the existing `post_search`, super cool!